docs/source/clickhouse/schema_design.rst

```diff
@@ -16,31 +18,37 @@ data in a column that contains two ``Nested`` arrays where the first array conta
 of the dictionary and the second array contains the values. This works well when
 your dataset design gives you the ability to filter based on other attributes to a small
 number of rows and the flexibility of completely arbitrary keys and values are a real requirement.
-Often, however, a ClickHouse user is interested in rows that contain a specific tag key or a
-specific ``key=value`` pair. In this case one is often best-served by creating a top-level
-column and "promoting" the data to it [#dupe]_.
-
-This promotion is preferable on a couple dimensions:
-
-1. Even though it looks like a dictionary, our tags and contexts "maps" are actually arrays of keys and values.
-   With array accesses Clickhouse needs to locate the index of the key whose value needs to be looked up.
-   That is an O(N) lookup. The same key may be located at different indexes for each row.
-   Once the index is known, looking up its value is O(1) since you perform direct index access.
-   If the query has to lookup M rows, this becomes an O(M * N) in terms of time complexity.
-
-   With direct column access, there is no index lookup for a key. The overall complexity to lookup M rows
-   for "promoted columns" is O(M).
-2. Queries that filter based on an array column may have to load the entire column into memory
-   in order to do filtering. Adding a new index on top of a top-level column is a more
-   straightforward operation that guarantees more consistent performance outcomes vs. iterating
-   over an array
+Often, however, a ClickHouse user is predominantly interested in rows that contain a specific tag key or a
+specific ``key=value`` pair that occurs in a very large percentage of the table data (for example,
+``http_status_code=200`` or ``hasKey(http_status_code)``). This breaks any efficiency
+of the bloom-filter style index on dictionary columns (see :ref:`bloom`).
```

Comment:
I am not sure I follow this paragraph. Does it mean that those conditions are not efficient with a bloom filter ?

It would be the other way around.
By default a nested column does not have a bloom filter index associated. 
That way, in order to find rows that contain a specific key-value pair require scanning the entire column and scanning all the tags for each row.

When we add the bloom filter index, we can make this search efficient as the bloom filter allows us to drop blocks that are very unlikely to contain `key=value` or `key` before scanning.

So the bloom filter index helps in these cases and only in these cases.

The bloom filter index is useless, though, in any other use of the nested column like:
- regex expressions
- key LIKE %something%
- arrayJoin(tags.value)

Reply:
Those conditions wouldn't be made efficient with a bloom filter index because (presumably) most of the rows in the hypothetical table have a `http_status_code` key and have `http_status_code=200` and therefore your BF will almost always be falling back to a table scan.

I do think that a mention of selectivity in queries like your below comment will maybe help make this clearer